### ISSUE TITLE
✅ 🐞 Locale with "comma" as decimal separator brakes generated CSS styles for decimal values

### ISSUE THREAD

**Issue by arnisjuraga** [elementor/elementor#10992]
I am using latest Elementor with Elementor Pro.

In php, set locale to the one with "comma as decimal seperator", like: Latvian "lv"  or "lv_LV"
(you must have this locale available in your server installation, for Ubuntu run
```apt-get install -y language-pack-lv``` )

an then put in themes  ```functions.php```
~~~php
<?php 
setlocale(LC_ALL, 'lv_LV.utf-8');
~~~

After this, ```.css``` files with decimal values will be generated with 'comma' as separators, and will be ignored by browsers as they semantically incorrect:

![image](https://user-images.githubusercontent.com/2185289/77762906-e7f58080-7042-11ea-99ed-4a36b0ce2f40.png)

I found these 2 similar issues but for much older Elementor, sot irrelevant I think:

#2300 
#2437

As temporary solution, put ```setlocale(LC_NUMERIC, 'en_US.UTF8');``` right after localized locale if used. This puts "dot" as decimal separator and assures correct .css generation. But still -0 should be done in Elementor plugin itself.

**Comment by jgruszka93** [comment]
Hi. I've got the same problem, I'm actually using polish locale setting, where we use comma as decimal separator. Elementor still prints commas into css files and that generates lots of errors. 

![image](https://user-images.githubusercontent.com/37851321/82837408-3ac7a900-9ec9-11ea-8d43-86a795fad211.png)

Your temporary solution doesn't work for me, or maybe i just dont know in which file i need to put `setlocale(LC_NUMERIC, 'en_US.UTF8');` Any advice?

**Comment by jgruszka93** [comment]
> Your temporary solution doesn't work for me, or maybe i just dont know in which file i need to put `setlocale(LC_NUMERIC, 'en_US.UTF8');` Any advice?

I fixed it- should be `setlocale(LC_NUMERIC, 'en_US.UTF-8');` 

However, this bug should be fixed in Elementor itself. In current version **Elementor is unusable in lots of locale settings that use comma as decimal separator.**

**Comment by tommargol** [comment]
hi jgruszka93 — could you tell me in which file I should place this code if using polish language?

**Comment by tommargol** [comment]
> > Your temporary solution doesn't work for me, or maybe i just dont know in which file i need to put `setlocale(LC_NUMERIC, 'en_US.UTF8');` Any advice?
> 
> I fixed it- should be `setlocale(LC_NUMERIC, 'en_US.UTF-8');`
> 
> However, this bug should be fixed in Elementor itself. In current version **Elementor is unusable in lots of locale settings that use comma as decimal separator.**

hi jgruszka93 — could you tell me in which file I should place this code if using polish language?

**Comment by jgruszka93** [comment]
> hi jgruszka93 — could you tell me in which file I should place this code if using polish language?

.\wp-content\themes\YOURTHEME\functions.php
.\wp-content\plugins\elementor\includes\controls\functions.php

That combination worked for me :)

**Comment by arnisjuraga** [comment]
Putting this in plugin files is not save, of course, it could be overwritten with plugin update. 
Putting it at the beginning of Your `functions.php` should be enough. 

Keep in mind - it does not matter, what language You are using. `LC_NUMERIC` should be set for `en_US`, because this locale is most available on servers and contains `.` as decimal separator. 

On some servers "en_US.UTF-8" can be missing, then it's enough to just use `en_US`.
And one more note from [Pigmeu](https://www.php.net/manual/en/function.setlocale.php#Hcom46640)
> The locale string need to be supported by the server. Sometimes there are diferents charsets for a language, like "pt_BR.utf-8" and "pt_BR.iso-8859-1", but there is no support for a _standard_ "pt_BR".


```setlocale(LC_NUMERIC, 'en_US');```"
```setlocale(LC_NUMERIC, 'en_US.UTF-8');```

**Comment by hubert-jaworski** [comment]
> > hi jgruszka93 — could you tell me in which file I should place this code if using polish language?
> 
> .\wp-content\themes\YOURTHEME\functions.php
> .\wp-content\plugins\elementor\includes\controls\functions.php
> 
> That combination worked for me :)

Does not work for me, sadly! Elementor no longer has the functions.php file in that folder.
Putting that snippet into the theme's functions.php (at the bottom or at the top) does not work.
Any ideas?

**Comment by jgruszka93** [comment]
> Does not work for me, sadly! Elementor no longer has the functions.php file in that folder.
> Putting that snippet into the theme's functions.php (at the bottom or at the top) does not work.
> Any ideas?

Hi Hubert. You're right, after last update there is no longer `functions.php` in Elementor. Putting 
```
<?php
setlocale(LC_NUMERIC, 'en_US.UTF-8');
```
at the beginning of  `.\wp-content\themes\YOURTHEME\functions.php` work for me. 

Have you tried both `setlocale(LC_NUMERIC, 'en_US');"` and `setlocale(LC_NUMERIC, 'en_US.UTF-8');` ?

**Comment by EssWF** [comment]
Hi!
The same problem and doesnt help for me
Any suggestions?

> Hi Hubert. You're right, after last update there is no longer `functions.php` in Elementor. Putting
> 
> ```
> <?php
> setlocale(LC_NUMERIC, 'en_US.UTF-8');
> ```
> 
> at the beginning of `.\wp-content\themes\YOURTHEME\functions.php` work for me.
> 
> Have you tried both `setlocale(LC_NUMERIC, 'en_US');"` and `setlocale(LC_NUMERIC, 'en_US.UTF-8');` ?

**Comment by CptSparky** [comment]
Hi there for me on nazwa.pl server worked adding to theme functions.php :
setlocale(LC_NUMERIC, 'en_US');
setlocale(LC_NUMERIC, 'en_US.UTF-8');
however it did not solved imediatly problem. I had to redit the page in elementor setting the values ones more. Only then when I saved the changes it worked for this particular page. 
My problem was with column widths which were custom set by me and this coma thing prevented them to display so they were displayed with default values. But now it is ok. Also I received this answer from nazwa.pl support but didn't have to try this solution:
Hello,

please add a variable

setlocale (LC_ALL, 'en_US.utf8');

in wp-config at the beginning or index.php at the very top in the code. That should fix the problem.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor  v3.17.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
